vector/src/main/java/im/vector/adapters/RoomDirectoryAdapter.java

```diff
@@ -0,0 +1,228 @@
+     * View holders
+     * *********************************************************************************************
+     */
+
+    class RoomDirectoryViewHolder extends RecyclerView.ViewHolder {
+        final View vMainView;
+        final ImageView vAvatarView;
+        final TextView vServerTextView;
+
+        private RoomDirectoryViewHolder(final View itemView) {
+            super(itemView);
+            vMainView = itemView;
+            vAvatarView = (ImageView) itemView.findViewById(R.id.room_directory_avatar);
+            vServerTextView = (TextView) itemView.findViewById(R.id.room_directory_display_name);
+        }
+
+        private void populateViews(final RoomDirectoryData server) {
+            vServerTextView.setText(server.getDisplayName());
+            setAvatar(vAvatarView, server.getAvatarUrl(), server.isIncludedAllNetworks() ? null : vServerTextView.getContext().getResources().getDrawable(R.drawable.network_matrix));
+
+            vMainView.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    mListener.onSelectRoomDirectory(server);
+                }
+            });
+        }
+    }
+
+    /*
+     * *********************************************************************************************
+     * avatar downloader
+     * *********************************************************************************************
+     */
+    private static final Map<String, Bitmap> mAvatarByUrl = new HashMap<>();
+
+    /**
+     * Load the avatar bitmap in the provided imageView
+     *
+     * @param imageView the image view
+     * @param avatarURL the avatar URL
+     */
+    private void setAvatar(final ImageView imageView, final String avatarURL, Drawable defaultAvatar) {
+        imageView.setImageDrawable(defaultAvatar);
+        imageView.setTag(null);
+
+        if (null != avatarURL) {
+            Bitmap bitmap = mAvatarByUrl.get(avatarURL);
+            imageView.setTag(avatarURL);
+
+            // if the image is not cached, download it
+            if (null == bitmap) {
+                new AsyncTask<Void, Void, Bitmap>() {
+                    @Override
+                    protected Bitmap doInBackground(Void... params) {
+                        Bitmap bitmap = null;
+                        try {
+                            URL url = new URL(avatarURL);
+                            bitmap = BitmapFactory.decodeStream((InputStream) url.getContent());
+                        } catch (Exception e) {
+                            Log.e(LOG_TAG, "## setAvatar() : cannot load the avatar " + avatarURL);
+                        }
+                        return bitmap;
+                    }
+
+                    @Override
+                    protected void onPostExecute(Bitmap bitmap) {
+                        if ((null != bitmap) && !mAvatarByUrl.containsKey(avatarURL)) {
```

Comment:
Check that imageView is not null? Maybe it may occur if the activity is closed before the asynctask is done.